src/lib_json/CMakeLists.txt

```diff
@@ -105,104 +75,62 @@ list(APPEND REQUIRED_FEATURES
-    endif()
-
-    set(SHARED_LIB ${PROJECT_NAME}_lib)
-    add_library(${SHARED_LIB} SHARED ${PUBLIC_HEADERS} ${JSONCPP_SOURCES})
-    set_target_properties(${SHARED_LIB} PROPERTIES
-        OUTPUT_NAME jsoncpp
-        VERSION ${PROJECT_VERSION}
-        SOVERSION ${PROJECT_SOVERSION}
-        POSITION_INDEPENDENT_CODE ${BUILD_SHARED_LIBS}
-    )
-
-    # Set library's runtime search path on OSX
-    if(APPLE)
-        set_target_properties(${SHARED_LIB} PROPERTIES INSTALL_RPATH "@loader_path/.")
-    endif()
-
-    target_compile_features(${SHARED_LIB} PUBLIC ${REQUIRED_FEATURES})
+# Property: OBJECT
 
-    target_include_directories(${SHARED_LIB} PUBLIC
-        $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
-        $<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/${JSONCPP_INCLUDE_DIR}>
-        $<BUILD_INTERFACE:${PROJECT_BINARY_DIR}/include/json>
-    )
 
-    list(APPEND CMAKE_TARGETS ${SHARED_LIB})
-endif()
-
-if(BUILD_STATIC_LIBS)
-    set(STATIC_LIB ${PROJECT_NAME}_static)
-    add_library(${STATIC_LIB} STATIC ${PUBLIC_HEADERS} ${JSONCPP_SOURCES})
-
-    # avoid name clashes on windows as the shared import lib is also named jsoncpp.lib
-    if(NOT DEFINED STATIC_SUFFIX AND BUILD_SHARED_LIBS)
-        if (MSVC)
-            set(STATIC_SUFFIX "_static")
-        else()
-            set(STATIC_SUFFIX "")
-        endif()
-    endif()
+add_library(jsoncpp.framework.object OBJECT
+        json_reader.cpp
+        json_value.cpp
+        json_writer.cpp
+        )
 
-    set_target_properties(${STATIC_LIB} PROPERTIES
-        OUTPUT_NAME jsoncpp${STATIC_SUFFIX}
-        VERSION ${PROJECT_VERSION}
-    )
+set_target_properties(jsoncpp.framework.object PROPERTIES CXX_STANDARD 11)
+set_target_properties(jsoncpp.framework.object PROPERTIES CXX_EXTENSIONS OFF)
+set_target_properties(jsoncpp.framework.object PROPERTIES CXX_STANDARD_REQUIRED ON)
+set_target_properties(jsoncpp.framework.object PROPERTIES VERSION ${PROJECT_VERSION})
+set_target_properties(jsoncpp.framework.object PROPERTIES SOVERSION ${PROJECT_SOVERSION})
 
-    # Set library's runtime search path on OSX
-    if(APPLE)
-        set_target_properties(${STATIC_LIB} PROPERTIES INSTALL_RPATH "@loader_path/.")
-    endif()
+target_compile_features(jsoncpp.framework.object PUBLIC ${REQUIRED_FEATURES})
+target_include_directories(jsoncpp.framework.object PUBLIC $<BUILD_INTERFACE:${JSONCPP_ROOT_DIR}/include>)
 
-    target_compile_features(${STATIC_LIB} PUBLIC ${REQUIRED_FEATURES})
 
-    target_include_directories(${STATIC_LIB} PUBLIC
-        $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
-        $<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/${JSONCPP_INCLUDE_DIR}>
-        $<BUILD_INTERFACE:${PROJECT_BINARY_DIR}/include/json>
-    )
+if (JSONCPP_BUILD_SHARED_LIBS)
 
-    list(APPEND CMAKE_TARGETS ${STATIC_LIB})
-endif()
+    # CMake link shared library on Windows
+    # Ref: https://stackoverflow.com/a/41618677
+    set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS TRUE)
```

Comment:
Better to make it an option rather than set forcefully

Reply:
In fact it is the only option that exists to obtain a similar behavior in both Linux and Windows without having to touch code or use the export directives in each symbol to be exported, this simulates the default behavior that occurs in Linux to export all symbols by default, which does not happen in Windows.

Although you are right, the code already has the JSON_API directives that solve this problem, I will add the option.

Reply:
I mean, option() with ON as default would allow user overwrite in case there's any compatibility issue with their setup.

Reply:
In my opinion we should not add an eventual fix to a hypothetical event, for my part it is more convenient to wait for the confirmation of the compatibility problem with the incident report.

Likewise, if you have an article where the compatibility issue with CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS is discussed, I would like to have a document about them to start mitigating the impact of the change.